### PR TITLE
generate_ptable.sh: support 8GB eMMC

### PR DESCRIPTION
**Pull request by hzhuang1**
Now there're three kinds of eMMC capacity size in HiKey. They're
3.56GB, 3.6GB & 7.8GB.

So we have to provide 4 different partition tables.
aosp4g -- AOSP partition table for both 3.56GB and 3.6GB eMMC.
aosp8g -- AOSP partition table for 7.8GB eMMC.
linux4g -- Debian partition table for both 3.56GB and 3.6GB eMMC.
linux8g -- Debian partition table for 7.8GB eMMC.

Signed-off-by: Haojian Zhuang haojian.zhuang@linaro.org

### CODE REVIEW

File: generate_ptable.sh [hunk 1]
```diff
@@ -13,9 +13,12 @@ case ${PTABLE} in
   tiny)
     SECTOR_NUMBER=81920
     ;;
-  aosp|linux)
+  aosp4g|linux4g)
```
Comment (fboudra) [inline generate_ptable.sh:16]: aosp_|linux_)

File: generate_ptable.sh [hunk 2]
```diff
@@ -29,7 +32,7 @@ case ${PTABLE} in
     sgdisk -n 1:2048:4095 -t 1:0700 -u 1:F9F21F01-A8D4-5F0E-9746-594869AEC3E4 -c 1:"vrl" -p ${TEMP_FILE}
     sgdisk -n 2:4096:6143 -t 2:0700 -u 2:F9F21F02-A8D4-5F04-9746-594869AEC3E4 -c 2:"vrl_backup" -p ${TEMP_FILE}
     ;;
-  aosp)
+  aosp4g|aosp8g)
```
Comment (fboudra) [inline generate_ptable.sh:35]: aosp*)

File: generate_ptable.sh [hunk 3]
```diff
@@ -53,7 +56,7 @@ case ${PTABLE} in
     #[10: userdata: 2126M-End]
     sgdisk -n -E -t 10:8300 -u 10:064111F6-463B-4CE1-876B-13F3684CE164 -c 10:"userdata" -p ${TEMP_FILE}
     ;;
-  linux)
+  linux4g|linux8g)
```
Comment (fboudra) [inline generate_ptable.sh:59]: linux*)

File: Makefile [hunk 4]
```diff
@@ -10,10 +10,14 @@ l-loader.bin: start.S debug.S
 	$(LD) -Bstatic -Tl-loader.lds -Ttext 0xf9800800 start.o debug.o -o loader
 	$(OBJCOPY) -O binary loader temp
 	python gen_loader.py -o l-loader.bin --img_loader=temp --img_bl1=bl1.bin
-	sudo PTABLE=aosp bash -x generate_ptable.sh
-	python gen_loader.py -o ptable-aosp.img --img_prm_ptable=prm_ptable.img
-	sudo PTABLE=linux bash -x generate_ptable.sh
-	python gen_loader.py -o ptable-linux.img --img_prm_ptable=prm_ptable.img
+	sudo PTABLE=aosp4g bash -x generate_ptable.sh
```
Comment (fboudra) [inline Makefile:13]: can we separate with dash: aosp-4g, aosp-8g, linux-4g, linux-8g

**Comment by fboudra** [comment]
Haojian, a few nitpick changes

**Comment by hzhuang1** [comment]
Close it. Use a new pull request instead. https://github.com/96boards/l-loader/pull/8

**Pull request closed without merging**